### PR TITLE
Curves supplier for dynawo

### PR DESCRIPTION
**Pull request by marcosmc**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the new behavior (if this is a feature change)?**
Implement the DSL to support the configuration of curves and the processor of the DSL

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
         uses: actions/checkout@v1	
         with:	
           repository: powsybl/powsybl-core	
-          ref: refs/heads/master	
+          ref: refs/heads/dynamic-curves-provider
```
Comment (agnesLeroy) [inline .github/workflows/maven.yml:23]: is this the correct branch name? It seems the branch is named curves_provider to me...

File: pom.xml [hunk 2]
```diff
@@ -69,6 +69,10 @@
                     </classpathDependencyExcludes>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.codehaus.gmavenplus</groupId>
+                <artifactId>gmavenplus-plugin</artifactId>
+            </plugin>
         </plugins>
     </build>
```
Comment (agnesLeroy) [inline pom.xml:77]: Why do you need this dependency?

File: src/main/java/com/powsybl/dynawo/Main.java [hunk 3]
```diff
@@ -53,7 +54,7 @@ public static void main(String[] args) {
         DynawoSimulationProvider provider = new DynawoSimulationProvider();
         provider.setDydFilename(dydFile);
         try (ComputationManager computationManager = new LocalComputationManager(LocalComputationConfig.load())) {
-            provider.run(network, computationManager, network.getVariantManager().getWorkingVariantId(), parameters).join();
+            provider.run(network, CurvesSupplier.empty(), computationManager, network.getVariantManager().getWorkingVariantId(), parameters).join();
```
Comment (agnesLeroy) [inline src/main/java/com/powsybl/dynawo/Main.java:57]: In the main we'll want to use a curves supplier now, so actually the main should be modified to make it possible to take the curves file as an argument and then run should be called with the corresponding curve supplier.

File: .github/workflows/maven.yml (hunk 1 above)
Comment (marcosmc) [reply .github/workflows/maven.yml:23]: Yes, it is the branch in powsybl-core with the implementation of the curve supplier made by Mathieu.

File: pom.xml (hunk 2 above)
Comment (marcosmc) [reply pom.xml:77]: This plugin is required to compile the DynawoCurveGroovyExtension.groovy class

File: src/main/java/com/powsybl/dynawo/Main.java (hunk 3 above)
Comment (marcosmc) [reply src/main/java/com/powsybl/dynawo/Main.java:57]: We split the step 2 of the roadmap in two parts, this PR is the fisrt one, in a second PR we will develop the curves exporter to an XML file, and modify the Main class to use the curves supplier.

File: src/main/groovy/com/powsybl/dynawo/simulator/DynawoCurveGroovyExtension.groovy [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            cloned()
+
+            if (curveSpec.staticId && curveSpec.modelId) {
+                throw new DslException("Both staticId and modelId are defined");
+            }
+            if (!curveSpec.variable) {
+                throw new DslException("'variable' field is not set")
+            }
+            if (curveSpec.staticId) {
+                consumer.accept(new DynawoCurve("NETWORK", curveSpec.staticId + "_" + curveSpec.variable));
```
Comment (rosiereflo) [inline src/main/groovy/com/powsybl/dynawo/simulator/DynawoCurveGroovyExtension.groovy:87]: What if this node is associated to a dynamic model?

Comment (zamarrenolm) [reply src/main/groovy/com/powsybl/dynawo/simulator/DynawoCurveGroovyExtension.groovy:87]: The corresponding dynamic model `modelId` can be given explicitly. In this case, an exception is raised if also an `staticId` is provided. 
So the user may refer to a variable in a known dynamic model or to an element for which no specific model has been given, and then should be accessible through `NETWORK` using its `staticId`.
Is this correct in Dynawo ?

Comment (rosiereflo) [reply src/main/groovy/com/powsybl/dynawo/simulator/DynawoCurveGroovyExtension.groovy:87]: Ok, seems correct, thanks!

File: src/main/java/com/powsybl/dynawo/simulator/DynawoSimulationProvider.java [hunk 5]
```diff
@@ -70,12 +71,12 @@ public String getVersion() {
...
 
     @Override
-    public CompletableFuture<DynamicSimulationResult> run(Network network, ComputationManager computationManager, String workingVariantId, DynamicSimulationParameters parameters) {
+    public CompletableFuture<DynamicSimulationResult> run(Network network, CurvesSupplier curvesSupplier, String workingVariantId, ComputationManager computationManager, DynamicSimulationParameters parameters) {
         Objects.requireNonNull(workingVariantId);
         Objects.requireNonNull(parameters);
 
         DynawoSimulationParameters dynawoParameters = getDynawoSimulationParameters(parameters);
-        return run(network, computationManager, workingVariantId, parameters, dynawoParameters);
+        return run(network, workingVariantId, computationManager, parameters, dynawoParameters);
```
Comment (mathbagu) [inline src/main/java/com/powsybl/dynawo/simulator/DynawoSimulationProvider.java:79]: Is something missing here? Why don't you use the curvesSupplier?

Comment (marcosmc) [reply src/main/java/com/powsybl/dynawo/simulator/DynawoSimulationProvider.java:79]: As there is no export to CRV XML in this PR, curvesSupplier is not used, in the next PR the list of curves obtained from the curvesSupplier is stored. If necessary, we can include it in this PR.

File: src/main/groovy/com/powsybl/dynawo/simulator/DynawoCurveGroovyExtension.groovy [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+        void staticId(String staticId) {
+            this.staticId = staticId
+        }
+
+        void variable(String variable) {
+            this.variable = variable
+        }
+    }
```
[the remark this answers is not in the document]
Comment (marcosmc) [reply src/main/groovy/com/powsybl/dynawo/simulator/DynawoCurveGroovyExtension.groovy]: Done

**Pull request merged**